packages/mobile/src/walletConnect/saga.ts

```diff
@@ -0,0 +1,306 @@
+      state: {
+        accounts: [`${account}@celo:${NETWORK_ID}`],
+      },
+    }
+
+    yield call(client.approve.bind(client), { proposal: session, response })
+  } catch (e) {
+    Logger.debug(TAG + '@acceptSession', e.message)
+  }
+}
+
+export function* denySession({ session }: DenySession) {
+  try {
+    if (!client) {
+      throw new Error('missing client')
+    }
+    yield call(client.reject.bind(client), {
+      reason: getError(WalletConnectErrors.NOT_APPROVED),
+      proposal: session,
+    })
+  } catch (e) {
+    Logger.debug(TAG + '@denySession', e.message)
+  }
+}
+
+export function* closeSession({ session }: CloseSession) {
+  try {
+    if (!client) {
+      throw new Error('missing client')
+    }
+    yield call(client.disconnect.bind(client), {
+      topic: session.topic,
+      reason: getError(WalletConnectErrors.USER_DISCONNECTED),
+    })
+  } catch (e) {
+    Logger.debug(TAG + '@closeSession', e.message)
+  }
+}
+
+export function* acceptRequest({
+  request: {
+    request: { id, jsonrpc, method, params },
+    topic,
+  },
+}: AcceptRequest): any {
+  try {
+    if (!client) {
+      throw new Error('Missing client')
+    }
+
+    const account: string = yield select(currentAccountSelector)
+    const wallet: UnlockableWallet = yield call(getWallet)
+
+    let result: any
+
+    if (method === SupportedActions.eth_signTransaction) {
+      yield call(unlockAccount, account)
+      result = (yield call(wallet.signTransaction.bind(wallet), params)) as EncodedTransaction
+    } else {
+      throw new Error('Unsupported action')
+    }
+
+    yield call(client.respond.bind(client), {
+      topic,
+      response: {
+        id,
+        jsonrpc,
+        result,
+      },
+    })
+  } catch (e) {
+    Logger.debug(TAG + '@acceptRequest', e.message)
+  }
+
+  const [, nextRequest] = yield select(selectPendingActions)
+  if (nextRequest) {
+    navigate(Screens.WalletConnectActionRequest, { request: nextRequest })
+    return
+  }
+
+  navigateBack()
+}
+
+export function* denyRequest({
+  request: {
+    request: { id, jsonrpc },
+    topic,
+  },
+}: DenyRequest) {
+  try {
+    if (!client) {
+      throw new Error('Missing client')
+    }
+
+    yield call(client.respond.bind(client), {
+      topic,
+      response: {
+        id,
+        jsonrpc,
+        error: getError(WalletConnectErrors.NOT_APPROVED),
+      },
+    })
+  } catch (e) {
+    Logger.debug(TAG + '@denyRequest', e.message)
+  } finally {
+    navigateBack()
+  }
+}
+
+export function* watchWalletConnectChannel() {
+  const walletConnectChannel: EventChannel<WalletConnectActions> = yield call(
+    createWalletConnectChannel
+  )
+  while (true) {
+    const message: WalletConnectActions = yield take(walletConnectChannel)
+    yield put(message)
+  }
+}
+
+export function* createWalletConnectChannel() {
+  if (!client) {
+    Logger.debug(TAG + '@initialiseClient', `init start`)
+    client = yield call(WalletConnectClient.init as any, {
```

Comment:
Can we remove `as any` here?

Reply:
I had issues with the types here, this seemed the least invasive way to go about it. 

We still know further on that `client` is a `WalletConnectClient`. Not sure that `call` plays so well with static functions. Could I also just do `yield WalletConnectClient.init(...)`?

Reply:
`yield call(...)` is the way to go here, otherwise it bypasses the normal redux saga flow.
Fine to keep `as any` for now.